### ISSUE TITLE
 Incompatible shapes: [1,249,311,2] vs. [1,249,311]

### ISSUE THREAD

**Issue by mumuyanyan** [davidsandberg/facenet#1103]
Caused by op 'pnet/sub', defined at:
  File "/home/nd/.pycharm_helpers/pydev/pydevd.py", line 1664, in <module>
    main()
  File "/home/nd/.pycharm_helpers/pydev/pydevd.py", line 1658, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/home/nd/.pycharm_helpers/pydev/pydevd.py", line 1068, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/nd/.pycharm_helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/compare.py", line 132, in <module>
    main(parse_arguments(sys.argv[1:]))
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/compare.py", line 43, in main
    images = load_and_align_data(args.image_files, args.image_size, args.margin, args.gpu_memory_fraction)
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/compare.py", line 92, in load_and_align_data
    pnet, rnet, onet = align.detect_face.create_mtcnn(sess, None)
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/align/detect_face.py", line 285, in create_mtcnn
    pnet = PNet({'data':data})
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/align/detect_face.py", line 73, in __init__
    self.setup()
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/align/detect_face.py", line 230, in setup
    .softmax(3,name='prob1'))
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/align/detect_face.py", line 51, in layer_decorated
    layer_output = op(self, layer_input, *args, **kwargs)
  File "/home/nd/workspace/workspace_lsh/3d/deep3dface_mac_tensor/facenet/src/align/detect_face.py", line 212, in softmax
    target_exp = tf.exp(target-max_axis)
  File "/home/nd/anaconda3/envs/deep3d/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 894, in binary_op_wrapper
    return func(x, y, name=name)
  File "/home/nd/anaconda3/envs/deep3d/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 4636, in _sub
    "Sub", x=x, y=y, name=name)
  File "/home/nd/anaconda3/envs/deep3d/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/nd/anaconda3/envs/deep3d/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
    op_def=op_def)
  File "/home/nd/anaconda3/envs/deep3d/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Incompatible shapes: [1,249,311,2] vs. [1,249,311]
	 [[Node: pnet/sub = Sub[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"](pnet/conv4-1/BiasAdd, pnet/Max)]]
